BlockEntity/BEFruitpress.cs

```diff
@@ -390,25 +418,59 @@ private bool InteractMashContainer(IPlayer byPlayer, BlockSelection blockSel)
-                    removeItems = 1;
+                    // the juiceableLitresLeft and juiceableLitresTransfered are per item if we have a stack of multiple
+                    // so we check if the press is full and only remove and add the liters for one item at a time
+                    if (juiceableLitresLeft + juiceableLitresTransfered + transferableLitres + usedLitres > juiceableLitresCapacity)
+                    {
+                        (Api as ICoreClientAPI)?.TriggerIngameError(this, "fullcontainer", Lang.Get("Container is full, press out juice and remove the mash before adding more"));
+                        return false;
+                    }
+
+                    // If we're adding mash we want to average the spoiling so players can keep it fresh forever
+                    TransitionState[] sourceTransitionStates = handStack.Collectible.UpdateAndGetTransitionStates(Api.World, handslot);
+                    TransitionState[] targetTransitionStates = mashStack.Collectible.UpdateAndGetTransitionStates(Api.World, MashSlot);
+
+                    Dictionary<EnumTransitionType, TransitionState> targetStatesByType = null;
+
+                    targetStatesByType = new Dictionary<EnumTransitionType, TransitionState>();
+                    foreach (var state in targetTransitionStates) targetStatesByType[state.Props.Type] = state;
+
+                    // We're mixing based on total litres because we don't really have a stack size to compare
+                    float t = (transferableLitres + usedLitres) / (transferableLitres + usedLitres + (float)juiceableLitresLeft + (float)juiceableLitresTransfered);
+
+                    foreach (var sourceState in sourceTransitionStates)
+                    {
+                        TransitionState targetState = targetStatesByType[sourceState.Props.Type];
+                        mashStack.Collectible.SetTransitionState(mashStack, sourceState.Props.Type, sourceState.TransitionedHours * t + targetState.TransitionedHours * (1 - t));
+                    }
+
+                    removeItems =  1;
                 } else
                 {
-                    float desiredTransferSizeLitres = byPlayer.Entity.Controls.ShiftKey ? (float)hprops.LitresPerItem : Math.Min(handStack.StackSize, 4) * (float)hprops.LitresPerItem;
-                    transferableLitres = (float)Math.Min(desiredTransferSizeLitres, juiceableLitresCapacity - juiceableLitresLeft);
+                    // In order to make sure we're always giving exactly the amount of juice that's appropriate for the
+                    // number of items we take out of the inventory we start by counting up how many the player wants to
+                    // add, be it 1, 4, or 32, and then subtracting 1 from that total until we have an amount that can fit
+                    int desiredTransferAmount = Math.Min(handStack.StackSize, byPlayer.Entity.Controls.ShiftKey ? 1 : byPlayer.Entity.Controls.CtrlKey ? 32 : 4);

```

Comment:
The change to BlockFruitpressTop fixes this so you can actually Shift + Click just 1 in.  I also added a Ctrl + Click 32 option just to speed things along.